hsmd/hsm.c

```diff
@@ -610,7 +610,7 @@ static void sign_funding_tx(struct daemon_conn *master, const u8 *msg)
 	struct bitcoin_tx *tx;
 	u16 outnum;
 	size_t i;
-	struct pubkey changekey;
+	struct pubkey changekey = {};
```

Comment:
Isn't this exactly the kind of initialization that @rustyrussell hates?

Reply:
I'm afraid so! :-(

What makes `funding_tx`, `list_for_each` and `list_for_each_safe` special? Seems like only these three trigger uninit warnings.

If `cppcheck` had a general issue with uninit `struct`:s being passed as arguments we'd see false positives all over the code base I guess?

Reply:
Yes!

In this case, it's close to correct.  We *do* hand changekey to funding_tx, but it's only used if change != 0: the same condition which we use to initialize changekey.

But I don't get the warning here with cppcheck 1.80 in Ubuntu.  I'll try frobbing it and seeing if I can fix it...